src/Client/Adapter/Curl.php

```diff
@@ -209,6 +209,13 @@ public function connect($host, $port = 80, $secure = false)
             }
         }
 
+        if (isset($this->config['sslcafile']) && $this->config['sslcafile']) {
```

Comment:
What about this alternative:
- if `sslverifypeer` is not set and `CURLOPT_CAINFO/CAPATH` are set, we assume that `sslverifypeer` is true
- if `sslverifypeer` is manually set to false, we don't consider `CURLOPT_CAINFO/CAPATH`